mailserver/limiter_test.go

```diff
@@ -0,0 +1,100 @@
+package mailserver
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+)
+
+func TestLimiterSuite(t *testing.T) {
+	suite.Run(t, new(LimiterSuite))
+}
+
+type LimiterSuite struct {
+	suite.Suite
+}
+
+func (s *LimiterSuite) TestIsAllowed() {
+	peerID := "peerID"
+	testCases := []struct {
+		t               time.Duration
+		shouldBeAllowed bool
+		db              func() map[string]time.Time
+		errMsg          string
+		info            string
+	}{
+		{
+			t:               5 * time.Millisecond,
+			shouldBeAllowed: true,
+			db: func() map[string]time.Time {
+				return make(map[string]time.Time)
+			},
+			errMsg: "Expected limiter not to allow with empty db",
+			info:   "Expecting limiter.isAllowed to not allow with an empty db",
+		},
+		{
+			t:               5 * time.Millisecond,
+			shouldBeAllowed: true,
+			db: func() map[string]time.Time {
+				db := make(map[string]time.Time)
+				db[peerID] = time.Now().Add(time.Duration(-10) * time.Millisecond)
+				return db
+			},
+			errMsg: "Expected limiter to allow with peer on its db",
+			info:   "Expecting limiter.isAllowed to allow with an expired peer on its db",
+		},
+		{
+			t:               5 * time.Millisecond,
+			shouldBeAllowed: false,
+			db: func() map[string]time.Time {
+				db := make(map[string]time.Time)
+				db[peerID] = time.Now().Add(time.Duration(-1) * time.Millisecond)
+				return db
+			},
+			errMsg: "Expected limiter to not allow with peer on its db",
+			info:   "Expecting limiter.isAllowed to not allow with a non expired peer on its db",
+		},
+	}
+
+	for _, tc := range testCases {
+		s.T().Logf(tc.info)
```

Comment:
i like to use sub-tests (`T().Run`), as it allows to filter by them. not sure if it is still possible with testify suites

Reply:
nice! yes they do, got it working with something like:
```
	for _, tc := range testCases {
		s.T().Run(tc.info, func(*testing.T) {
			s.server.limit = nil
			err := s.server.Init(s.shh, &tc.config)
			s.server.tick = nil
			s.server.Close()
			s.Equal(tc.expectedError, err)
			s.Equal(tc.limiterActive, (s.server.limit != nil))
		})
	}
```